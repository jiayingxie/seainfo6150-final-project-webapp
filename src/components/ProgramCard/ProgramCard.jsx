import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import styles from './ProgramCard.module.css';

export default class ProgramCard extends Component {
    render() {
        const { department, programID, programName } = this.props;
        return (
            <li className={styles.container}>
                <div className={styles.card}>
                    <h2>{programID}</h2>
                    <h4 className={styles.link}><Link to={`/engineering/${department}/${programID}`}>{programName}</Link></h4>
                </div>
            </li>
        )
    }
}
