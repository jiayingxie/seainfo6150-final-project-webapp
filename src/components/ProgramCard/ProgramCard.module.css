.container{
    background-image: linear-gradient(to right, rgba(102,169,201, 0.8), #eee 20%);
    border: 1px solid #668B8B;
    border-radius: 10px;
    box-shadow: 0 5px 10px 5px rgba(0, 0, 0, 0.3);
    font-family: Arial, Helvetica, sans-serif;
    padding: 10px;
}

.card {
    display: flex;
    justify-items: center;
    flex-wrap: wrap;
    justify-content: space-between;
    height: 100%;
}

.card h2 {
    width: 100%;
    font-family: Georgia, Times, sans-serif;
    margin: 10px 0;
}

.card h4 {
    margin: 10px 0;
}

.card .link {
    width: 100%;
}