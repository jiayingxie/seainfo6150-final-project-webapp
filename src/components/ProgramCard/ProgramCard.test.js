import React from "react";
import ProgramCard from "./ProgramCard";

describe("ProgramCard tests", () => {
    let department;
    let program;

    // the method is to get test data
    const fetchData = async () => {
        const mul = "multidisciplinary";
        const CSECS = "CSECS";
        const response = await fetch(`https://demo2247515.mockable.io/engineering`);
        const responseJson = await response.json();
        department = await responseJson[mul];
        program = await Object.values(department.programs).find(prog => { return prog.programID === CSECS });
    };
    it("renders correctly when passed a department prop and program props", async () => {
        await fetchData();
        const { container } = render(<ProgramCard department={department} {...program} />);
        expect(container).toMatchSnapshot();
    });
});