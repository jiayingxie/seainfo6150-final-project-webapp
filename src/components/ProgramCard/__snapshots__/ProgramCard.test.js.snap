// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgramCard tests renders correctly when passed a department prop and program props 1`] = `
<div>
  <li
    class="container"
  >
    <div
      class="card"
    >
      <h2>
        CSECS
      </h2>
      <h4
        class="link"
      >
        <a
          href="/engineering/multidisciplinary/CSECS"
        >
          Computer Systems Engineering with Concentration in Software Design Engineering
        </a>
      </h4>
    </div>
  </li>
</div>
`;
